AK/StdLibExtras.h

```diff
@@ -119,6 +119,13 @@ constexpr T mix(T const& v1, T const& v2, U const& interpolation)
     return v1 + (v2 - v1) * interpolation;
 }
 
+template<typename T, typename U>
+constexpr T smooth_step(T edge0, T edge1, U x)
+{
+    x = clamp((x - edge0) / (edge1 - edge0), 0.0f, 1.0f);
+    return x * x * (3 - 2 * x);
+}
```

Comment:
As far as I know, "the" smooth step function is a primarily graphics (shader) driven function that isn't commonly part of any standard library that I know of. Incorporating this in LibGfx might be fine though?

Reply:
I placed it there because we have other functions that feel similar there (like `mix()` and `clamp()`, and such)

Reply:
Things like interpolating/mixing and clamping are much more ubiquitous and definitely not limited to graphics, whereas I've never seen smoothstep outside of graphics processing.

Reply:
I'll just move this to be a lambda where it's used for now (and we can find a home for if it's ever needed elsewhere).